src/EnergyPlus/DataHeatBalance.hh

```diff
@@ -2178,6 +2178,18 @@ namespace DataHeatBalance {
         {
         }
     };
+    struct HeatReclaimHPCoilData
+    {
+        // Members
+        std::string Name;       // Name of WAHP Coil
+        std::string SourceType; // SourceType for WAHP Coil
+        Real64 AvailCapacity;   // Total available heat reclaim capacity
+
+        // Default Constructor
+        HeatReclaimHPCoilData() : AvailCapacity(0.0)
```

Comment:
For the structure, it is because there seems to be no water to air heat pump system heat reclaim data structure existing, so maybe it's a good idea to have one to store data for that kind of systems. So far I just followed the DX structure and the way it models desuperheater, so it looks exactly the same, but maybe for other coils or systems it would add more features which I am not sure.

And the name and source types I didn't quite understand the question. But in the DXCoils.cc, the name, source type in HeatReclaimDXCoilData are initialized when processing the input data after allocated to the specific coil object.

Reply:
The point here is that if the elements of the new struct HeatReclaimHPCoilData are the same as HeatReclaimDXCoilData then just do this and you won't need the new struct for HeatReclaimHPCoilData:

Array1D<HeatReclaimDXCoilData> HeatReclaimSimple_WAHPCoil;

Reply:
Yes I knew, so far they're exactly the same thing. We could use that DX structure, it would work but the name would look weird, do we want to change the name? It might be included in multiple namespaces.
Or actually the new structure could be added some differences. Before, I returned the calculated reclaimed heat by referring the ``WaterThermalTanks.cc`` namespace in ``WaterToAirHeatPumpSimple.cc`` to calculate the reduction of heat transfer to ground. Another way maybe we could do it is to pass that information through the new data structure. 
Any thought?

Reply:
@yzhou601 That sounds reasonable to me. My only outstanding recommendation on this would be to initialize `Name` and `SourceType` to something in your default constructor on this line. Not a big deal, but it's good to not leave ambiguity unnecessarily.

Reply:
Sure, I will take care of it

Reply:
@Myoldmopar are we initializing strings now in the constructor?

Reply:
For the record, yes, this is a great opportunity to **not** just add in a brand new structure that looks identical to another one.  This would have been addressed in a design document review process, as a statement that said "duplicate structure A" would have been declined.  If we have multiple structures here that look almost identical, then we should very quickly extract out a common base class and create derived types that only add their own customization.  It literally shouldn't be more than a 15 minute job to do it for this new type and the one it was duplicated from, and would go a long way toward improving maintenance as opposed to hindering it.

@rraustad As for initialization, we have been doing initializer-list as the norm for a while, and it's generally fine, but we don't do a whole lot of initialization during our program, so there's not a significant performance difference for us.  I actually prefer the readability of inline initialization, so putting `int i = 1;` in the class definition, rather than `i(1)` in the constructor initializer list.  But I haven't been picky when reviewing yet until we settle on a preferred form.